.github/workflows/prestocpp-macos-build.yml

```diff
@@ -0,0 +1,41 @@
+name: prestocpp-macos-build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - 'presto-native-execution/**'
+
+jobs:
+  prestocpp-macos-build-engine:
+    runs-on: macos-13
+    steps:
+      - uses: actions/checkout@v4
+      - name: Fix git permissions
+        # Usually actions/checkout does this but as we run in a container
+        # it doesn't work
+        run: git config --global --add safe.directory ${GITHUB_WORKSPACE}
+      - name: Update submodules
+        run: |
+          cd presto-native-execution
+          make submodules
+      - uses: maxim-lobanov/setup-xcode@v1.6.0
```

Comment:
Can you comment on why we need this? Is it to improve build time?
Can we use macos15? https://github.com/actions/runner-images/tree/main/images/macos

Reply:
If you look at the spec for the runners: https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

you can see that macOS 14 and 15 (the latter in beta) has only 7GB memory.
It already is a slow build and only having 7GB memory would really impact it. I think we should wait and change if we get better runners for newer macOS versions.

In addition, we want to build with Clang15 so we need to install a newer version of XCode.

I deliberately chose macOS13 with XCode 15 which is why we need this for the above reasons.

Reply:
Can you also add a comment in the yml file?